client/my-sites/domains/domain-management/settings/cards/glue-records-card.tsx

```diff
@@ -164,7 +164,11 @@ export default function GlueRecordsCard( { domain }: { domain: ResponseDomain }
 		if ( ! nameserver ) {
 			return false;
 		}
-		if ( ! nameserver.match( /^[A-Za-z0-9](?:[A-Za-z0-9-]{0,61}[A-Za-z0-9])?$/ ) ) {
+		if (
+			! nameserver.match(
+				/^([A-Za-z0-9](?:[A-Za-z0-9-]{0,61}[A-Za-z0-9])?)(.[A-Za-z0-9](?:[A-Za-z0-9-]{0,61}[A-Za-z0-9])?)*$/
```

Comment:
```suggestion
				/^([A-Za-z0-9](?:[A-Za-z0-9-]{0,61}[A-Za-z0-9])?)(\.[A-Za-z0-9](?:[A-Za-z0-9-]{0,61}[A-Za-z0-9])?)*$/
```
I think we need a backslash before the period there 🤔

Reply:
Also, I was testing long hostnames and it seems there's a 50 character limit in total for these name servers 😅 anything longer than that returns an `Invalid attribute value` error from KS. Couldn't find anything about this in the docs 😞 

So I guess we can add a `nameserver.length > 50` check before the regex check.

Reply:
Good catch! Thanks @leonardost!